hashintel/hash
# Align ESLint scripts in `package.json`, replace `.eslintignore` with `ignorePatterns`

## 🌟 What is the purpose of this PR?

This PR continues to improve ESLint setup in the monorepo. The main goal of this PR is to align ESLint feedback in CI (CLI) with what we see in the editor. This is achieved via changes in `.eslintrc.cjs` files as well as the replacement of `.eslintignore` with [`ignorePatterns`](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files) (which are generated from `.prettierignore` and are workspace-specific).




## 🔗 Related links





- [Asana task 1](https://app.asana.com/0/1203007126736613/1203268787004306/f) _(internal)_
- [Asana task 2](https://app.asana.com/0/1203007126736613/1203268787004304/f) _(internal)_
## 🚫 Blocked by

- [x] https://github.com/hashintel/hash/pull/1319

## 🔍 What does this change?




See inline diff comments for context

## 📜 Does this require a change to the docs?

No

## ⚠️ Known issues

Linting rules have been carried over, which means that the setup is far from perfect. Improving rules is out of this PR's scope.

## 🐾 Next steps

Test new ESLint setup, continue improving it


## 🛡 What tests cover this?



- CI
- Manual experiments in VSCode

## ❓ How to test this?

- Check CI
- Check https://github.com/blockprotocol/blockprotocol/pull/733 (blocks still build)
- Open VSCode, reload ESLint extension and check feedback, e.g. by adding

  ```ts
  console.log("X");
  let x = 42;
  ```
